### PR TITLE
upstreaming and porting 2018 Sienna dbc. file

### PR DESCRIPTION
**Pull request by hyperactive32**
https://community.comma.ai/cabana/?route=3b6e00dde85c3462%7C2018-12-20--00-35-11&max=5&url=https%3A%2F%2Fchffrprivate.blob.core.windows.net%2Fchffrprivate3%2Fv2%2F3b6e00dde85c3462%2Ff7b244e071e181247d50f587ed06ec60_2018-12-20--00-35-11

### CODE REVIEW

**Comment by vanillagorillaa** [comment]
Looks like you guys have two PR's open for the Sienna. You will want to close one of them

**Comment by vanillagorillaa** [comment]
Also better to have the DBC files in a separate PR in the opendbc repo. You will also want to add the sienna to the Readme chart

**Comment by vanillagorillaa** [comment]
@hyperactive32 Add the Sienna to the README in the supported cars list, as well as fix the conflict in the values.py file and should be good

**Comment by vanillagorillaa** [comment]
Also, I think @rbiasini will want the DBC portion of the port to be it's own PR in openDBC.

File: selfdrive/car/toyota/interface.py [hunk 1]
```diff
@@ -140,6 +140,16 @@ def get_params(candidate, fingerprint):
...
       ret.steerKf = 0.00006
+      
+    elif candidate == CAR.SIENNA:
+      ret.safetyParam = 100
+      ret.wheelbase = 3.03 # 119.3in =  3030.22mm
+      ret.steerRatio = 16.0
+      tire_stiffness_factor = 0.444 # not optimized yet
+      ret.mass = 4590 * CV.LB_TO_KG + std_cargo #mean between normal (4505) and limited (4750)
+      ret.steerKpV, ret.steerKiV = [[0.18], [0.0075]]
+      ret.steerKf = 0.0003
```
Comment (rbiasini) [inline selfdrive/car/toyota/interface.py]: this tuning seems pretty weird :). You sure that it's better than, for example, the one from the Highlander?

Comment (hyperactive32) [reply selfdrive/car/toyota/interface.py]: it's not as jerky with this tune but still needs corner tuning

Comment (hyperactive32) [reply selfdrive/car/toyota/interface.py]: @vanillagorillaa what values errors?

Comment (hyperactive32) [reply selfdrive/car/toyota/interface.py]: Added to readme. working on values now

**Comment by icmma** [comment]
@rbiasini  did you get chance to look into static messages that it had?

**Comment by rbiasini** [comment]
@icmma, for us to verify the static messages I need:
- cabana route with stock system running
- cabana route with camera unplugged
- cabana route with camera AND DSU unplugged

**Comment by rbiasini** [comment]
@icmma @hyperactive32 
What's the state of this PR? Added few issues I could quickly spot in static messages.

File: selfdrive/car/toyota/values.py [hunk 2]
```diff
@@ -22,39 +22,39 @@ class ECU:
...
+  (0x240, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH, CAR.SIENNA), 1,   5, '\x00\x10\x01\x00\x10\x01\x00'),
+  (0x241, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH, CAR.SIENNA), 1,   5, '\x00\x10\x01\x00\x10\x01\x00'),
+  (0x244, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH, CAR.SIENNA), 1,   5, '\x00\x10\x01\x00\x10\x01\x00'),
+  (0x245, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH, CAR.SIENNA), 1,   5, '\x00\x10\x01\x00\x10\x01\x00'),
+  (0x248, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH, CAR.SIENNA), 1,   5, '\x00\x00\x00\x00\x00\x00\x01'),
+  (0x367, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH, CAR.SIENNA), 0,  40, '\x06\x00'),
+  (0x414, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH, CAR.SIENNA), 0, 100, '\x00\x00\x00\x00\x00\x00\x17\x00'),
```
Comment (rbiasini) [inline selfdrive/car/toyota/values.py]: I don't see 0x414 in the route you posted. 0x412 is there instead.
I also see 0x488 being sent by stock camera and not present here.

File: selfdrive/car/toyota/values.py [hunk 3]
```diff
@@ -22,39 +22,39 @@ class ECU:
...
   (0x466, ECU.CAM, (CAR.COROLLA), 1, 100, '\x24\x20\xB1'),
-  (0x489, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH), 0, 100, '\x00\x00\x00\x00\x00\x00\x00'),
-  (0x48a, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH), 0, 100, '\x00\x00\x00\x00\x00\x00\x00'),
-  (0x48b, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH), 0, 100, '\x66\x06\x08\x0a\x02\x00\x00\x00'),
+  (0x489, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH, CAR.SIENNA), 0, 100, '\x00\x00\x00\x00\x00\x00\x00'),
+  (0x48a, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH, CAR.SIENNA), 0, 100, '\x00\x00\x00\x00\x00\x00\x00'),
+  (0x48b, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA, CAR.HIGHLANDER, CAR.HIGHLANDERH, CAR.SIENNA), 0, 100, '\x66\x06\x08\x0a\x02\x00\x00\x00'),
   (0x4d3, ECU.CAM, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA), 0, 100, '\x1C\x00\x00\x01\x00\x00\x00\x00'),
```
Comment (rbiasini) [inline selfdrive/car/toyota/values.py]: 0x4d3 is being sent by stock camera

**Comment by icmma** [comment]
not sure where adam is but there is new user with sienna will be able to get more data from him long as he doesn't go dark lol

**Comment by rbiasini** [comment]
@hyperactive32 , is there an updated Sienna port? Tnx.

**Comment by icmma** [comment]
@rbiasini  i did help couple of users one of them had 19. I do have the fingerprints but the statics messages issue still remains. Im not able to contact them any longer any who following are 2019. 

`    36: 8, 37: 8, 114: 5, 119: 6, 120: 4, 170: 8, 180: 8, 186: 4, 426: 6, 452: 8, 464: 8, 466: 8, 467: 8, 544: 4, 545: 5, 548: 8, 550: 8, 552: 4, 562: 4, 608: 8, 610: 5, 643: 7, 705: 8, 725: 2, 764: 8, 800: 8, 824: 8, 835: 8, 836: 8, 849: 4, 869: 7, 870: 7, 888: 8, 896: 8, 900: 6, 902: 6, 905: 8, 911: 8, 916: 1, 918: 7, 921: 8, 933: 8, 944: 6, 945: 8, 951: 8, 955: 8, 956: 8, 979: 2, 992: 8, 998: 5, 999: 7, 1000: 8, 1001: 8, 1002: 8, 1008: 2, 1014: 8, 1017: 8, 1041: 8, 1043: 8, 1056: 8, 1059: 1, 1076: 8, 1077: 8, 1114: 8, 1164: 8, 1165: 8, 1166: 8, 1167: 8, 1176: 8, 1177: 8, 1178: 8, 1179: 8, 1180: 8, 1181: 8, 1182: 8, 1183:8, 1191: 8, 1192: 8, 1196: 8, 1197: 8, 1198: 8, 1199: 8, 1200: 8, 1201: 8, 1202: 8, 1203: 8, 1212: 8, 1227: 8, 1228: 8, 1237: 8, 1279: 8, 1552: 8, 1553: 8, 1555: 8, 1556: 8, 1557: 8, 1561: 8, 1562: 8, 1568: 8, 1569: 8, 1570: 8, 1571: 8, 1572: 8, 1584: 8, 1589: 8, 1592: 8, 1593: 8, 1595: 8, 1656: 8, 1664: 8, 1666: 8, 1667: 8, 1728: 8, 1745: 8, 1779: 8`

**Comment by icmma** [comment]
If you want i can merge it to latest.

**Comment by hyperactive32** [comment]
That's up to you I just moved and have not found time to update mine as of
yet.

**Comment by rbiasini** [comment]
Sienna is supported since 0.6.1

**Pull request closed without merging**